src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -205,18 +206,15 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
                 return null;
             }
 
-            if (_requestMethod == RequestMethod.Post)
```

Comment:
I thought in the course of discussion @lmazuel said we should leave this check lest we fall into difficult cases. Or was that really only for `Azure-AsyncOperation` as you made changes for below?